dashboard/test/models/pd/regional_partner_mapping_test.rb

```diff
@@ -0,0 +1,36 @@
+require 'test_helper'
+
+class Pd::RegionalPartnerMappingTest < ActiveSupport::TestCase
+  test 'state must be in list' do
+    mapping = build :pd_regional_partner_mapping, state: 'invalid'
+    refute mapping.valid?
+    assert_equal ['State is not included in the list'], mapping.errors.full_messages
+  end
+
+  test 'zip code must be a valid format' do
+    mapping = build :pd_regional_partner_mapping, zip_code: 'invalid', state: nil
+    refute mapping.valid?
+    assert_equal ['Zip code is invalid'], mapping.errors.full_messages
+  end
+
+  test 'either zip code or state must be present but not both' do
+    mapping = create :pd_regional_partner_mapping
```

Comment:
nit: `build` instead of `create` since we're not leveraging the DB in this test.

Reply:
Or you could remove this entirely and generate new models (in memory) for each row in the test matrix.

```ruby
mapping = build :pd_regional_partner_mapping, zip_code: zip_code, state: state
```

instead of line 32 below

Reply:
Done